qualtran/bloqs/qsp/generalized_qsp.py

```diff
@@ -274,16 +340,51 @@ def decompose_from_registers(
             yield self.U.adjoint().on_registers(**quregs)
 
     def build_call_graph(self, ssa: 'SympySymbolAllocator') -> Set['BloqCountT']:
-        degree = len(self.P)
+        degree = len(self.P) - 1
 
-        counts = {(rotation, 1) for rotation in self.signal_rotations}
+        counts = set(Counter(self.signal_rotations).items())
 
         if degree > self.negative_power:
-            counts.add((self.U.controlled(control_values=[0]), degree - self.negative_power))
+            counts.add((Controlled(self.U, CtrlSpec(cvs=0)), degree - self.negative_power))
         elif self.negative_power > degree:
             counts.add((self.U.adjoint(), self.negative_power - degree))
 
         if self.negative_power > 0:
-            counts.add((self.U.adjoint().controlled(), min(degree, self.negative_power)))
+            counts.add((Controlled(self.U.adjoint(), CtrlSpec()), min(degree, self.negative_power)))
 
         return counts
+
+
+@bloq_example
+def _gqsp() -> GeneralizedQSP:
+    from qualtran.bloqs.for_testing.atom import TestGWRAtom
```

Comment:
I think we should use the gates defined in `for_testing/` only in tests. For examples, it's better to use commonly used basic gates; maybe add a `Identity` bloq if you need it for the example or use a `ZPow` or `XPow` gate ?

Reply:
Ah yes that makes sense, I'll just use `XGate` here.